torch/utils/cpp_extension.py

```diff
@@ -282,6 +282,28 @@ def _join_sycl_home(*paths) -> str:
     '-D__HIP_NO_HALF_CONVERSIONS__=1',
 ]
 
+_COMMON_SYCL_FLAGS = [
+    '-fsycl',
+    '-fsycl-targets=spir64_gen,spir64',
+]
+
+def _get_sycl_arch_list():
+    if 'TORCH_XPU_ARCH_LIST' in os.environ:
+        return os.environ.get('TORCH_XPU_ARCH_LIST')
+    arch_list = torch.xpu.get_arch_list()
```

Comment:
```suggestion
    if not torch.xpu.is_available():
        return ""
    arch_list = torch.xpu.get_arch_list()
```

Reply:
While this change will probably resolve the failure noticed by ci, it binds compiling of sycl extension to the systems which have XPU devices unless `TORCH_XPU_ARCH_LIST` is explicitly set.

I guess failure is coming from this line:
https://github.com/pytorch/pytorch/blob/dd5d0ea6bb058e84625a05b0531650d890824b30/torch/xpu/__init__.py#L423
I.e. at the moment `torch.xpu.get_arch_list()` will work only if torch is compiled with XPU support. I suggest to modify `torch.xpu.get_arch_list()` in a way that it will return `[]` in such a case instead. This will help to avoid having XPU device to compile sycl extensions.

Reply:
I will follow up on this proposal in a separate PR.

Reply:
> I will follow up on this proposal in a separate PR.

Submitted https://github.com/pytorch/pytorch/pull/147431